app/models/miq_widget.rb

```diff
@@ -492,6 +492,10 @@ def self.sync_from_hash(attrs)
     widget
   end
 
+  def filter_for_schedule
+    %["=" => {"field" => "MiqWidget-id", "value" => #{id}}]
```

Comment:
Is this supposed to be a string or a hash?  It looks like it used to be a hash

Reply:
I missed that when I looked at this earlier. `MiqExpression` will not complain if a string is passed on creation. But it will blow up when it gets evaluated. It needs to be a hash.

Reply:
yeah, i missed it, old rules - every linr of new code should go with unit testing ...
wil fix it and add rspec too